certora/ArrangerConduit.spec

```diff
@@ -0,0 +1,805 @@
+rule drawFunds(address asset, address destination, uint256 amount) {
+    env e;
+
+    require asset == gem;
+    require currentContract != destination;
+    
+    mathint balanceOfConduitBefore = gem.balanceOf(currentContract);
+    mathint balanceOfDestinationBefore = gem.balanceOf(destination);
+
+    require balanceOfConduitBefore + balanceOfDestinationBefore <= max_uint256;
+
+    drawFunds(e, asset, destination, amount);
+
+    mathint balanceOfConduitAfter = gem.balanceOf(currentContract);
+    mathint balanceOfDestinationAfter = gem.balanceOf(destination);
+
+    assert balanceOfConduitAfter == balanceOfConduitBefore - amount, "balance of conduit did not decrease by amount";
+    assert balanceOfDestinationAfter == balanceOfDestinationBefore + amount, "balance of destination did not increase by amount";
+}
+
+// Verify revert rules on drawFunds
+rule drawFunds_revert(address asset, address destination, uint256 amount) {
+    env e;
+
+    require asset == gem;
+    require currentContract != destination;
+    
+    mathint balanceOfConduit = gem.balanceOf(currentContract);
+    mathint totalWithdrawableFunds = totalWithdrawableFunds(asset);
+    
+    bool isBroker = isBroker(destination, asset);
+    address arranger = arranger();
+
+    drawFunds@withrevert(e, asset, destination, amount);
+
+    bool revert1 = e.msg.value > 0;
+    bool revert2 = arranger != e.msg.sender;
+    bool revert3 = totalWithdrawableFunds > balanceOfConduit;
+    bool revert4 = to_mathint(amount) > balanceOfConduit - totalWithdrawableFunds;
+    bool revert5 = !isBroker;
+    bool revert6 = balanceOfConduit < to_mathint(amount);
+
+    assert lastReverted <=> revert1 || revert2 || revert3 ||
+                            revert4 || revert5 || revert6
+                            , "Revert rules failed";
+
+}
+
+// Verify correct storage changes for non reverting returnFunds
+rule returnFunds(uint256 fundRequestId, uint256 returnAmount) {
+    env e;
+
+    uint256 anyIndex;
+
+    IArrangerConduit.FundRequest requestBefore = getFundRequest(anyIndex);
+    bytes32 infoHashBefore = aux.hashString(requestBefore.info);
+
+    mathint requestedFundsBefore = requestedFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalRequestedFundsBefore = totalRequestedFunds(requestBefore.asset);
+    mathint withdrawableFundsBefore = withdrawableFunds(requestBefore.asset, requestBefore.ilk);
+    mathint totalWithdrawableFundsBefore = totalWithdrawableFunds(requestBefore.asset);
+    mathint numRequestsBefore = getFundRequestsLength();
+
+    bytes32 otherIlk;
+    address otherAsset;
+    require otherIlk != requestBefore.ilk || otherAsset != requestBefore.asset;
+    mathint requestedFundsOtherBefore = requestedFunds(otherAsset, otherIlk);
+
+    address otherAsset2;
+    require otherAsset2 != requestBefore.asset;
+    mathint totalRequestedFundsOtherBefore = totalRequestedFunds(otherAsset2);
+
+    bytes32 otherIlk2;
```

Comment:
Same here, with `otherIlk`, `otherAsset` and `otherAsset2` should be enough for all the variables.

Reply:
Done